src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,157 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class handles saving and parsing saved task files.
+ */
+public class Storage {
+    private final String filePath;
+
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Get the file handle of the save file, or create the directories and the save file if it does not exist.
+     *
+     * @return A File object referencing the save file.
+     * @throws DukeException if the directories or file cannot be created.
+     */
+    private File getOrCreateFile() throws DukeException {
+        Path path = Paths.get(this.filePath);
+        try {
+            Files.createDirectories(path.getParent());
+        } catch (IOException e) {
+            throw new DukeException(e.getMessage());
+        }
+
+        File file = new File(filePath);
+        if (!file.exists()) {
+            try {
+                Files.createFile(path);
+            } catch (IOException e) {
+                throw new DukeException(e.getMessage());
+            }
+        }
+
+        return file;
+    }
+
+    /**
+     * Load the tasks from the save file.
+     *
+     * @return An ArrayList containing all the tasks.
+     * @throws DukeException if there is an error accessing the file, or if the file data was corrupted.
+     */
+    public ArrayList<Task> loadTasks() throws DukeException {
+        File file = this.getOrCreateFile();
+        String regexPattern = "([TDE])\\s\\|\\s"    // match type
```

Comment:
Consider declaring constants for recurring values like the regex pattern to enhance maintainability further.